### ISSUE TITLE
NetworkInterface.GetAllNetworkInterfaces() in AWS Lambda throws DirectoryNotFoundException - .netcore 3.1

### ISSUE THREAD

**Issue by pawanrao** [dotnet/runtime#50293]
### Description

The method `NetworkInterface.GetAllNetworkInterfaces()` always throws a `DirectoryNotFoundException` if executed in an AWS Lambda environment.

Reproducer: 
```c#
using System.Net.NetworkInformation;

[assembly: Amazon.Lambda.Core.LambdaSerializer(typeof(Amazon.Lambda.Serialization.Json.JsonSerializer))]
namespace TestLambda
{
    public class Function
    {
        public void FunctionHandler(Amazon.Lambda.Core.ILambdaContext context)
        {
            // Here we get a "DirectoryNotFoundException"
            var adapters = NetworkInterface.GetAllNetworkInterfaces();
	    Console.Writeline("Successfully loaded network interfaces");
        }
    }
}

```
Exception: 
```
One or more errors occurred. (Could not find a part of the path '/sys/class/net/lo/mtu'.) (Could not find a part of the path '/sys/class/net/telemetry1_sb/mtu'.) (Could not find a part of the path '/sys/class/net/vinternal_1/mtu'.) (Could not find a part of the path '/sys/class/net/vtarget_1/mtu'.) (Could not find a part of the path '/sys/class/net/lo/mtu'.) (Could not find a part of the path '/sys/class/net/telemetry1_sb/mtu'.) (Could not find a part of the path '/sys/class/net/vinternal_1/mtu'.) (Could not find a part of the path '/sys/class/net/vtarget_1/mtu'.): AggregateException
Could not find a part of the path '/sys/class/net/lo/mtu'.: DirectoryNotFoundException
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream.OpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.StreamReader.ValidateArgsAndOpenPath(String path, Encoding encoding, Int32 bufferSize)
   at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks)
   at System.IO.File.InternalReadAllText(String path, Encoding encoding)
   at System.IO.File.ReadAllText(String path)
   at System.Net.NetworkInformation.StringParsingHelpers.ParseRawIntFile(String filePath)
   at System.Net.NetworkInformation.LinuxIPv4InterfaceProperties.GetMtu()
   at System.Net.NetworkInformation.LinuxIPv4InterfaceProperties..ctor(LinuxNetworkInterface linuxNetworkInterface)
   at System.Net.NetworkInformation.LinuxIPInterfaceProperties..ctor(LinuxNetworkInterface lni)
   at System.Net.NetworkInformation.LinuxNetworkInterface..ctor(String name, Int32 index)
   at System.Net.NetworkInformation.LinuxNetworkInterface.GetOrCreate(Dictionary`2 interfaces, String name, Int32 index)
   at System.Net.NetworkInformation.LinuxNetworkInterface.<>c__DisplayClass5_0.<GetLinuxNetworkInterfaces>b__2(String name, LinkLayerAddressInfo* llAddr)
```
### Configuration
.NET Core 3.1
AWS Lambda
linux-x64

### Regression?
https://github.com/dotnet/runtime/issues/23731
Seems to indicate that the issue was fixed in .netcore 2.1 but it is still reproducible.

**Comment by wfurt** [comment]
I don't think #23731 is complete fix - at least it did make exceptions catchable. 
GetMTU could still throw and there is no code to catch exception like some other places. 
 
https://github.com/dotnet/corefx/blob/29cfd61ab26fcf803f8f59641112e0fc39e10f7f/src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPv4InterfaceProperties.cs#L61-L65

It would be interesting oof you can try it with .NET 5.0 @pawanrao

**Comment by pawanrao** [comment]
@wfurt looks like this works fine with .NET5.0

Would it be possible to fix this in  .NETCore3.1 since that is the current LTS version?

**Comment by wfurt** [comment]
Thanks for confirmation @pawanrao. I think 3.1 would probably need different fix e.g. preserve existing implementation and fix the IO exception. (5.0 was somewhat major rewrite) 
Perhaps @karelz can comment on the 3.1 ask.

**Comment by karelz** [comment]
@pawanrao we would consider fixing 3.1 if there are multiple customers impacted by the bug without a workaround and with strong case for not being able to upgrade to .NET 5.0.
For a starter, what blocks you to upgrade to .NET 5?

In this particular case, we would have to be also convinced we can fix it in a targeted manner in 3.1 -- which is not clear now based on above assessment.

**Comment by pawanrao** [comment]
@karelz  We are hesitant to upgrade to a non-LTS version of .NET Core. However, if bugs like this are not fixed in LTS versions, then perhaps LTS doesn't mean what I thought it did.  Additionally, we host this particular app in AWS Lambda.  We would have to use containers to run .net 5, and currently, there are some AWS Lambda ECR container restrictions that preclude us from going that route. 
#23731 and #15806 lists various other reports of what appears to be the same bug(going back all the way to .NET Core 2)

**Comment by karelz** [comment]
@pawanrao can you share more details why you are hesitant to upgrade to non-LTS version?

If we turn LTS (or any servicing train) into "fix any bug anyone hits and considers interesting", we will ultimately hurt stability of that release (by making more changes, we increase risk of introducing new regressions -- which noone wants). That's why we look for signal from customers that it is really needed to be serviced.

I am not familiar with AWS Lambda -- @normj may be able to help here. Not sure why containers are not usable in your case. Is there an option to deploy self-contained app or is that out of question?
For context: Related info about .NET 5 support: https://github.com/aws/aws-lambda-dotnet/issues/727

Regarding the duplicates (#23731 and #15806) -- the bug was supposed to be fixed in .NET Core 2.0 (in PR https://github.com/dotnet/corefx/pull/13067), which shipped on 2017/8 (3.5 years ago).
This is the first complaint since then, so I wonder if it truly hurts all those developers (Does it just work for them? Did they work around it? Did they give up?), or if it is problematic only in certain corner cases ... do you (or @normj) have more thoughts about that?

**Comment by karelz** [comment]
Triage: No response for 2 weeks, closing for now. We need more info (see above) to justify backport to 3.1.
